### PR TITLE
Add default label in branch select list

### PR DESCRIPTION
**Pull request by yp05327**
image

### CODE REVIEW

File: web_src/js/components/RepoBranchTagSelector.vue [hunk 1]
```diff
@@ -30,6 +30,7 @@
         <div class="loading-indicator is-loading" v-if="isLoading"/>
         <div v-for="(item, index) in filteredItems" :key="item.name" class="item" :class="{selected: item.selected, active: active === index}" @click="selectItem(item)" :ref="'listItem' + index">
           {{ item.name }}
+          <div class="ui label" v-if="index===0 && mode === 'branches'">{{ textDefault }}</div>
```
Comment (lunny) [inline web_src/js/components/RepoBranchTagSelector.vue]: So we need to ensure the first branch is the default one?

Comment (yp05327) [reply web_src/js/components/RepoBranchTagSelector.vue]: https://github.com/go-gitea/gitea/blob/b62c8e7765a371600a300f62da96483a1ae0c731/routers/web/repo/repo.go#L648

emmm, it seems that in sometimes there's no default branch?

Comment (lunny) [reply web_src/js/components/RepoBranchTagSelector.vue]: No. I mean we need more comments both the two places.

File: web_src/js/components/RepoBranchTagSelector.vue [hunk 2]
```diff
@@ -30,6 +30,10 @@
         <div class="loading-indicator is-loading" v-if="isLoading"/>
         <div v-for="(item, index) in filteredItems" :key="item.name" class="item" :class="{selected: item.selected, active: active === index}" @click="selectItem(item)" :ref="'listItem' + index">
           {{ item.name }}
+          <!-- in backend, we put default branch as the first item, so just check item index is enought here -->
+          <div class="ui label" v-if="index===0 && mode === 'branches'">
```
Comment (wxiaoguang) [inline web_src/js/components/RepoBranchTagSelector.vue]: This assumption is too fragile, IMO the right approach is to let the backend tell frontend which is the default branch.

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -138,6 +138,8 @@ confirm_delete_selected = Confirm to delete all selected items?
 name = Name
 value = Value
 
+default = Default
```
Comment (wxiaoguang) [inline options/locale/locale_en-US.ini]: I do not think it is a general translation, especially when it is used in a label.

For example, GitHub show "default" but not "Default"

Comment (silverwind) [reply options/locale/locale_en-US.ini]: I guess it's fine, but could be reinforced by a key like `default_branch_default`.

File: web_src/js/components/RepoBranchTagSelector.vue (hunk 2 above)
Comment (yp05327) [reply web_src/js/components/RepoBranchTagSelector.vue]: Done.

File: options/locale/locale_en-US.ini (hunk 3 above)
Comment (yp05327) [reply options/locale/locale_en-US.ini]: Did some changes in https://github.com/go-gitea/gitea/pull/26697/commits/284d6449fcae8f1f2c9f54bedd657a9a850d99f9

**Review by lunny — APPROVED** [review]

**Review by delvh — APPROVED** [review]

**Pull request merged**